tests/unit/xtriggers/test_workflow_state.py

```diff
@@ -87,6 +90,19 @@ def test_back_compat(tmp_run_dir):
     satisfied, _ = workflow_state(id_, task='arkenstone', point='2012')
     assert not satisfied
 
+    # Test warning is shown:
+    from cylc.flow.xtriggers.suite_state import suite_state
+    assert (
+        'The suite_state xtrigger is deprecated.'
+        ' Please use the workflow_state xtrigger instead.' in
+        caplog.messages
+    )
+    caplog.clear()
+
+    monkeypatch.setattr('cylc.flow.flags.cylc7_back_compat', True)
+    importlib.reload(cylc.flow.xtriggers.suite_state)
+    assert caplog.messages == []
```

Comment:
I don't think this part of the test makes sense. @wxtim What was the rationale behind this?
```suggestion
```

Reply:
Quite right @MetRonnie  - this is the wrong way around: But we should just as much test that we don't get an error if the flag is False as do get the error if flag is true.

Reply:
Monkeypatch definitely works (there are other tests that patch the `cylc7_back_compat` flag). But testing this is a bit icky in pytest due to the warning happening on import. I suggest a functional test would be more appropriate (would have a quick runtime as it would just call validate a couple of times)

Reply:
OK, that's more reasonable. Although we are still starting extra Python instances.

Reply:
https://github.com/markgrahamdawson/cylc-flow/pull/4